### PR TITLE
Bump MSRV to 1.81.0

### PR DESCRIPTION
**Pull request by landonxjames**
## Checklist

- [ ] For changes to the smithy-rs codegen or runtime crates, I have created a changelog entry Markdown file in the `.changelog` directory, specifying "client," "server," or both in the `applies_to` key.
- [ ] For changes to the AWS SDK, generated SDK code, or SDK runtime crates, I have created a changelog entry Markdown file in the `.changelog` directory, specifying "aws-sdk-rust" in the `applies_to` key.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: tools/ci-scripts/codegen-diff/diff_lib.py [hunk 1]
```diff
@@ -55,12 +55,12 @@ def generate_and_commit_generated_code(revision_sha, targets=None, preserve_aws_
     # Move generated code into codegen-diff/ directory
     get_cmd_output(f"rm -rf {OUTPUT_PATH}")
     get_cmd_output(f"mkdir {OUTPUT_PATH}")
-    if target_aws_sdk in targets:
-        # Compiling aws-config for semver checks baseline requires build artifacts to exist under aws/sdk/build
-        if preserve_aws_sdk_build:
-            get_cmd_output(f"cp -r aws/sdk/build/aws-sdk {OUTPUT_PATH}/")
-        else:
-            get_cmd_output(f"mv aws/sdk/build/aws-sdk {OUTPUT_PATH}/")
+    # if target_aws_sdk in targets:
```
Comment (aajtodd) [inline tools/ci-scripts/codegen-diff/diff_lib.py:58]: no longer needed or what?

File: tools/ci-scripts/codegen-diff/semver-checks.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def main(skip_generation=False):
 
     if not skip_generation:
         checkout_commit_and_generate(head_commit_sha, CURRENT_BRANCH, targets=['aws:sdk'])
-        checkout_commit_and_generate(base_commit_sha, BASE_BRANCH, targets=['aws:sdk'], preserve_aws_sdk_build=True)
+        checkout_commit_and_generate(base_commit_sha, BASE_BRANCH, targets=['aws:sdk'], preserve_aws_sdk_build=False)
```
Comment (aajtodd) [inline tools/ci-scripts/codegen-diff/semver-checks.py:31]: intentional?

**Review by aajtodd — APPROVED** [review]

File: tools/ci-scripts/codegen-diff/diff_lib.py (hunk 1 above)
Comment (landonxjames) [reply tools/ci-scripts/codegen-diff/diff_lib.py:58]: Both of the bits you commented on were early attempts to fix the doubled crates issue in the semver check, neither change actually made any difference so I can revert them

Comment (landonxjames) [reply tools/ci-scripts/codegen-diff/diff_lib.py]: Reverted in https://github.com/smithy-lang/smithy-rs/pull/3936/commits/5f0c533be2a6ac951b9084b6853a5bea6aa2baaa

File: tools/ci-scripts/codegen-diff/semver-checks.py (hunk 2 above)
Comment (landonxjames) [reply tools/ci-scripts/codegen-diff/semver-checks.py]: Reverted in https://github.com/smithy-lang/smithy-rs/pull/3936/commits/5f0c533be2a6ac951b9084b6853a5bea6aa2baaa

**Pull request merged**